vars/setJobDescriptionFromMetadata.groovy

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/groovy
+
+/**
+ * setJobDescriptionFromMetadata() step.
+ */
+def call(Map params = [:]) {
+    def pipelineMetadata = params.get('pipelineMetadata')
+
+    def jobDescription
+    def job = Jenkins.instance.getItemByFullName(env.JOB_NAME)
+
+    def description = pipelineMetadata.get('pipelineDescription')
+    def maintainer = pipelineMetadata.get('maintainer')
+    def docs = pipelineMetadata.get('docs')
+
+    def contact = ''
+    // translate 'contact' map into a human readable string, e.g.:
+    // [irc: '#fedora-ci', email: 'ci@lists.fedoraproject.org']
+    // translates into "via irc on #fedora-ci or via email on ci@lists.fedoraproject.org"
+    pipelineMetadata.get('contact').each { key, value ->
+        if (contact) {
+            contact += " or "
+        }
+        contact += "via ${key} on ${value}"
+    }
+
+    jobDescription = """
+<p>
```

Comment:
Afaik we don't have HTML enabled in Job and build descriptions on our Jenkins master instances.

Generally it is not enabled by default as it add security risks. Do you think it is required?

If so I think we need to start tracking this requirement.

Reply:
I didn't realize we don't have HTML enabled on our instances. I tested this on the CentOS instance and it worked.

Jenkins supports "safe-html" in few places, descriptions included. It's basically a "safe" subset of HTML tags: https://wiki.jenkins.io/display/JENKINS/Markup+formatting